public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -546,6 +552,30 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
     return false;
   }
 
+  async getDerivedFields(row: { [key: string]: string | number | null }): Promise<DerivedField[]> {
```

Comment:
TODO: needs to be fixed, see Loki datasource with better implementation.

Reply:
wonder what the best function signature is for this function, maybe DataFrame and row index? but not 100% sure this needed. If it is needed maybe something like getDerivedFields(data: DataFrame, rowIndex: number): DataFrame